File: Packets/src/main/java/me/ImSpooks/core/packets/collection/database/PacketRequestData.java
package me.ImSpooks.core.packets.collection.database;

import lombok.Getter;
import me.ImSpooks.core.packets.collection.GlobalPacket;
import me.ImSpooks.core.packets.init.Packet;
import me.ImSpooks.core.packets.init.channels.WrappedInputStream;
import me.ImSpooks.core.packets.init.channels.WrappedOutputStream;

import java.io.IOException;

/**
 * Created by Nick on 09 okt. 2019.
 * Copyright © ImSpooks
 */
@Getter
@GlobalPacket
public class PacketRequestData extends Packet {

    private String collection;
    private String key;
    private Object value;

    public PacketRequestData(String collection, String key, Object value) {
        this.collection = collection;
        this.key = key;
        this.value = value;
    }

    public PacketRequestData() {
    }

    @Override
    public void send(WrappedOutputStream out) throws IOException {
        out.writeString(this.collection);
        out.writeString(this.key);
        out.writeTypePrefixed(this.value);
    }

    @Override
    public void receive(WrappedInputStream in) throws IOException {
        this.collection = in.readString();
        this.key = in.readString();
        this.value = in.readTypePrefixed();
    }
}

File: ClientBridge/src/main/php/client/packets/collection/database/PacketRequestData.php
<?php
/**
 * Created by Nick on 25 okt 2019.
 * Copyright © ImSpooks
 */

namespace client\packets\collection\database;

use client\packets\init\Packet;
use client\packets\init\channels\WrappedInputStream;
use client\packets\init\channels\WrappedOutputStream;

class PacketRequestData extends Packet {

    /** @var string */ 	private $collection;
    /** @var string */ 	private $key;
    private $value;

    public function __construct(string $collection = "", string $key = "", $value = null) {
        $this->collection = $collection;
        $this->key = $key;
        $this->value = $value;
    }

    public function receive(WrappedInputStream $in) {
        $this->collection = $in->readString();
        $this->key = $in->readString();
        $this->value = $in->readTypePrefixed();
    }

    public function send(WrappedOutputStream $out) {
        $out->writeString($this->collection);
        $out->writeString($this->key);
        $out->writeTypePrefixed($this->value);
    }

    public function getCollection(): string {
        return $this->collection;
    }

    public function getKey(): string {
        return $this->key;
    }

    public function getValue() {
        return $this->value;
    }

}
